aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1001,6 +1111,9 @@ static void apply_svd(Tensor& self, Tensor& U, Tensor& S, Tensor& VT,
   auto m = self.size(-2);
   auto n = self.size(-1);
   auto k = std::min(m, n);
+  int64_t lrwork = jobz == 'N' ? 5*k : k*std::max(5*k+7,2*std::max(m,n)+2*k+1);
```

Comment:
Where did this formula come from?

Reply:
Sorry for the confusion. I meant to take this settings from the LAPACK docs, but I think I copied the wrong setting when I was searching other sites for the reasoning behind this setting. The code has been updated with the setting for LAPACK 3.8.

The rwork setting in [LAPACK](http://www.netlib.org/lapack/explore-html/d3/da8/group__complex16_g_esing_gaccb06ed106ce18814ad7069dcb43aa27.html) is the minimum setting. Increasing this number by 2x, 10x, and 100x did not impact performance in an measurable way.